source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -506,9 +510,11 @@ bool CompressorFilterConfig::DirectionConfig::isMinimumContentLength(
     }
     return is_minimum_content_length;
   }
-
-  return StringUtil::caseFindToken(headers.getTransferEncodingValue(), ",",
-                                   Http::Headers::get().TransferEncodingValues.Chunked);
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.enable_compression_without_chunked_header")) {
+    return true;
```

Comment:
Please put a comment here that we are going to return true to account for HTTP/2

Reply:
Is it worth only doing this in the case of HTTP/2? Or should we do it for HTTP/1 also? I don't feel strongly about it but was wondering if the RFC says anything about this one way or the other?

Reply:
Basically the RFC says that any combination of `transfer-encoding: chunked` and `content-length` is valid. If both are present then the latter must be ignored. The lack of both is valid too with one exception: "a user agent that sends a request containing a message body MUST send a valid Content-Length header field if it does not know the server will handle HTTP/1.1 (or later) requests".  This is a rudiment of HTTP/1.0.

So, the current code in master disables compression for messages that are valid, but look suspicious. And in case of HTTP/2 there is no ground for any suspicions.

Reply:
I think we could figure out a way to determine if the request is HTTP/1 or HTTP/2, but if both are valid in both cases this is fine with me. Thanks!